pkg/cloudprovider/aws/instance.go

```diff
@@ -285,6 +276,22 @@ func (p *InstanceProvider) updateUnavailableOfferingsCache(ctx context.Context,
 	}
 }
 
+func (p *InstanceProvider) getCapacityType(constraints *v1alpha1.Constraints, instanceTypes []cloudprovider.InstanceType) string {
+	// Use spot if constraints are flexible to it and there is an offering for
+	// it. The AWS Cloud Provider defaults to [ on-demand ], so spot must be
+	// explicitly included in capacity type requirements.
```

Comment:
Nit: Should this go above the function instead of inside of it?

Anyway, good catch with this fix! I had thought of this case when working on my capacity type PR and then it totally slipped my mind when actually implementing ICE caching so I'm glad you thought of it.